### ISSUE TITLE
Handle failed put with a retry 

### ISSUE THREAD

**Issue by ryanramage** [ryanramage/couch2elastic4sync#4]
Our ES server sometimes gives us this garbage:

```
{"name":"couch2elastic4sync","hostname":"4ebbf0a5866f","pid":17,"level":30,"change":72957,"msg":"success.  update E4005687 2-af625df0664a8fb99c48d7b10d631470 null","time":"2016-02-25T15:52:31.314Z","v":0}
{"name":"couch2elastic4sync","hostname":"4ebbf0a5866f","pid":17,"level":50,"msg":"error occured update E4006947 2-b325dddeb63b2ab84c51c190a27eb4a3 { [Error: socket hang up] code: 'ECONNRESET' }","time":"2016-02-25T15:52:33.356Z","v":0
```

Flaky networking. While we fix the root of this problem, our little sync should handle retries better.

**Comment by ryanramage** [comment]
https://github.com/caolan/async#retry

**Comment by ryanramage** [comment]
fixed in https://github.com/ryanramage/couch2elastic4sync/pull/6, released 3.2.0.
